pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -276,6 +295,45 @@ func aContainerHasFailed(statuses []corev1.ContainerStatus) bool {
 	return false
 }
 
+func getScanResult(r *ReconcileComplianceScan, instance *complianceoperatorv1alpha1.ComplianceScan, node *corev1.Node) (complianceoperatorv1alpha1.ComplianceScanStatusResult, error) {
+	podName := fmt.Sprintf("%s-%s-pod", instance.Name, node.Name)
+	p := &corev1.Pod{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: podName, Namespace: instance.Namespace}, p)
+	if err != nil {
+		return complianceoperatorv1alpha1.ResultFailure, err
+	}
+	for _, status := range p.Status.ContainerStatuses {
+		if status.Name == OpenSCAPScanContainerName {
+			if status.State.Terminated != nil {
+				switch status.State.Terminated.ExitCode {
+				case 0:
+					return complianceoperatorv1alpha1.ResultCompliant, nil
+				case 2:
+					return complianceoperatorv1alpha1.ResultNonCompliant, nil
+				default:
+					return complianceoperatorv1alpha1.ResultFailure, fmt.Errorf(status.State.Terminated.Message)
+				}
+			} else {
+				return complianceoperatorv1alpha1.ResultFailure, fmt.Errorf("The pod '%s' was missing 'terminated' status", p.Name)
+			}
+		}
+	}
+	return complianceoperatorv1alpha1.ResultFailure, fmt.Errorf("Couldn't find '%s' container in '%s' pod status", OpenSCAPScanContainerName, p.Name)
+}
+
+func gatherResults(r *ReconcileComplianceScan, instance *complianceoperatorv1alpha1.ComplianceScan, nodes corev1.NodeList) (complianceoperatorv1alpha1.ComplianceScanStatusResult, error) {
+	var err error
+	var result complianceoperatorv1alpha1.ComplianceScanStatusResult
+	for _, node := range nodes.Items {
+		result, err = getScanResult(r, instance, &node)
+		if result != complianceoperatorv1alpha1.ResultCompliant {
+			// we output the last result if it was a failure
```

Comment:
Would it be more defensive if we, instead of using the last non-compliant status used the 'strongest' one si that a failure would trump non-compliance?

Reply:
Right, that would be a good idea. However, how do you output the strongest one? Another thing to note is that this iterates until it finds a non-compliant check. Which really would be all of the nodes, given that they should all have similar configurations (if we do a profile per node). So, there might really be no use in going through all of them, since one failure would in theory represent the whole set of nodes.

Reply:
By the strongest one, I meant that a failure would override non-compliance. I agree that in theory the results should all be the same, though. So I'm not sure if it is needed to cherry-pick the status.

Reply:
ah! I understand now... I'm honestly not sure. I'm open on this one. What do you think is best in this case? Error is should take precedence over non-compliance for the status?

Reply:
I made errors take precedence. Any thoughts?